runtime/client_http_response.go

```diff
@@ -66,8 +67,43 @@ func (res *ClientHTTPResponse) setRawHTTPResponse(httpRes *http.Response) {
 	res.Header = httpRes.Header
 }
 
+// readAndCaptureBody reads bytes from response and stores it in ClientHTTPResponse's rawResponseBytes
+func (res *ClientHTTPResponse) readAndCaptureBody() error {
+	// mark response read to true
+	defer func() {
+		res.responseRead = true
+	}()
+	rawBody, err := io.ReadAll(res.rawResponse.Body)
```

Comment:
can we make this a fucntion will be able to capture the same logic in file below

Reply:
updated the code, and made it more simple. Invoking ReadAll() method instead introducing new method